DxDispatch/src/dxdispatch/Executor.cpp

```diff
@@ -137,6 +137,20 @@ void Executor::operator()(const Model::DispatchCommand& command)
         return;
     }
 
+    ID3D12Resource* m_timestampResultBuffers;
+
+    const uint32_t timestampCount = m_device->timestampCount;
+
+    m_device->D3D()->CreateCommittedResource(
+        &CD3DX12_HEAP_PROPERTIES(D3D12_HEAP_TYPE_READBACK),
+        D3D12_HEAP_FLAG_NONE,
+        &CD3DX12_RESOURCE_DESC::Buffer(sizeof(UINT64) * 2 * timestampCount),
+        D3D12_RESOURCE_STATE_COPY_DEST,
+        nullptr,
+        IID_PPV_ARGS(&m_timestampResultBuffers));
```

Comment:
You need to store the results in a ComPtr (not raw pointer) or you'll leak memory. You can also use the existing helper:

```cpp
auto timestampResultsBuffer = m_device->CreateReadbackBuffer(sizeof(uint64_t) * timestampCount * 2)
```

Also, it's a little weird to store 2x timestampCount. Why not make timestampCount the actual number of timestamps instead of the number of iterations, effectively?

Reply:
I found it more intuitive at first to store the count of actual timings we can track, but it makes just as much sense to just follow the name and store the number of resources in this case.